odilo.py

```diff
@@ -0,0 +1,598 @@
+    def record_info_to_metadata(cls, book, availability):
+        """Turn Odilo's JSON representation of a book into a Metadata
+        object.
+
+        Note:  The json data passed into this method is from a different file/stream
+        from the json data that goes into the book_info_to_circulation() method.
+        """
+        if 'id' not in book:
+            return None
+
+        odilo_id = book['id']
+        primary_identifier = IdentifierData(Identifier.ODILO_ID, odilo_id)
+        active = book.get('active')
+
+        title = book.get('title')
+        subtitle = book.get('subtitle')
+        series = book.get('series')
+        series_position = book.get('seriesPosition')
+
+        contributors = []
+        author = book.get('author')
+        if author:
+            roles = [Contributor.AUTHOR_ROLE]
+            contributor = ContributorData(sort_name=author, display_name=author, roles=roles, biography=None)
+            contributors.append(contributor)
+
+        publisher = book.get('publisher')
+
+        # Metadata --> Marc21 260$c
+        published = book.get('publicationDate')
+        if not published:
+            # yyyyMMdd --> record creation date
+            published = book.get('releaseDate')
+
+        if published:
+            try:
+                published = datetime.datetime.strptime(published, "%Y%m%d")
+            except ValueError as e:
+                cls.log.warn('Cannot parse publication date from: ' + published + ', message: ' + e.message)
+
+        # yyyyMMdd --> record last modification date
+        last_update = book['modificationDate']
+        if last_update:
+            try:
+                last_update = datetime.datetime.strptime(last_update, "%Y%m%d")
+            except ValueError as e:
+                cls.log.warn('Cannot parse last update date from: ' + last_update + ', message: ' + e.message)
+
+        language = book.get('language')
+
+        subjects = []
+        for subject in book.get('subjects', []):
+            subjects.append(SubjectData(type=Subject.TAG, identifier=subject, weight=100))
+
+        grade_level = book.get('gradeLevel')
+        if grade_level:
+            subject = SubjectData(type=Subject.GRADE_LEVEL, identifier=grade_level, weight=10)
+            subjects.append(subject)
+
+        medium = None
+        formats = []
+        for format_received in book.get('formats', []):
+            # Both formats (PDF and EPUB) behave the same, and are basically the same thing for the final users.
+            if format_received in ('PDF', 'EPUB'):
+                format_received = 'EBOOK_STREAMING'
+            if format_received in cls.format_data_for_odilo_format:
+                content_type, drm_scheme = cls.format_data_for_odilo_format.get(format_received)
+                formats.append(FormatData(content_type, drm_scheme))
+                if not medium:
+                    medium = cls.odilo_medium_to_simplified_medium.get(format_received)
```

Comment:
It probably doesn't matter but feel like this should be outside the `if` statement in line 470.

Reply:
I think it's ok, because if 'format_received' it's in 'cls.format_data_for_odilo_format', implies that it's also in 'cls.odilo_medium_to_simplified_medium.get' and the other way round.